src/components/ModalAddToken.js

```diff
@@ -40,25 +60,52 @@ class ModalAddToken extends React.Component {
   }
 
   /**
-   * Method called when user clicks the button to register the token  
+   * Method called when user clicks the button to register the token
    * Validates that the data written is valid
    *
    * @param {Object} e Event emitted when user clicks the button
    */
-  handleAdd = (e) => {
+  handleAdd = async (e) => {
     e.preventDefault();
+
+    // Validating input field contents
     if (this.refs.config.value === '') {
       this.setState({ errorMessage: t`Must provide configuration string or uid, name, and symbol` });
       return;
     }
-    const promise = hathorLib.tokens.validateTokenToAddByConfigurationString(this.refs.config.value, null);
-    promise.then((tokenData) => {
-      tokens.addToken(tokenData.uid, tokenData.name, tokenData.symbol);
+
+    try {
+      const tokenData = await hathorLib.tokens.validateTokenToAddByConfigurationString(this.refs.config.value, null);
+      const tokensBalance = this.props.tokensBalance;
+
+      const tokenUid = tokenData.uid;
+      const tokenBalance = tokensBalance[tokenUid];
+      const tokenHasZeroBalance = (tokenBalance.available + tokenBalance.locked) === 0;
+
+      /*
+       * We only make this validation if the "Hide Zero-Balance Tokens" setting is active,
+       * and only do it once. If the warning message was shown, we will accept the "alwaysShow"
+       * checkbox value as the user decision already.
+       */
+      if (
+        wallet.areZeroBalanceTokensHidden()
+        && tokenHasZeroBalance
+        && !this.state.shouldExhibitAlwaysShowCheckbox
+      ) {
+        this.setState({
+          shouldExhibitAlwaysShowCheckbox: true,
+          errorMessage: t`This token has no balance on your wallet and you have the "hide zero-balance tokens" settings on.\nDo you wish to always show this token? (You can always undo this on the token info screen.)`
```

Comment:
I like this message, I just feel the UI is not good with this showing as an error in red. I'd rather create a new `<div>` similar to the error message with a warning message and use here.

What do you think?

Reply:
On commit [cbcea64](https://github.com/HathorNetwork/hathor-wallet/pull/310/commits/cbcea64a1fc82e11234c97855e2d3d7f51340f22) we improved this layout to the one below:

![Screenshot from 2022-05-18 19-58-43](https://user-images.githubusercontent.com/1299409/169168934-9338eb8c-83be-42c7-90fb-6568b5eb3655.png)